content/docs/composition-vs-inheritance.md

```diff
@@ -17,65 +17,19 @@ Some components don't know their children ahead of time. This is especially comm
 
 We recommend that such components use the special `children` prop to pass children elements directly into their output:
 
-```js{4}
-function FancyBorder(props) {
-  return (
-    <div className={'FancyBorder FancyBorder-' + props.color}>
-      {props.children}
-    </div>
-  );
-}
-```
+`embed:composition-vs-inheritance/fancy-border-example.js`
 
 This lets other components pass arbitrary children to them by nesting the JSX:
 
-```js{4-9}
-function WelcomeDialog() {
-  return (
-    <FancyBorder color="blue">
-      <h1 className="Dialog-title">
-        Welcome
-      </h1>
-      <p className="Dialog-message">
-        Thank you for visiting our spacecraft!
-      </p>
-    </FancyBorder>
-  );
-}
-```
-
-[Try it on CodePen.](https://codepen.io/gaearon/pen/ozqNOV?editors=0010)
+`embed:composition-vs-inheritance/welcome-dialog-example.js`
+
+[Try it on CodeSandbox.](codesandbox://composition-vs-inheritance/fancyborder-welcome-dialog-example)
 
 Anything inside the `<FancyBorder>` JSX tag gets passed into the `FancyBorder` component as a `children` prop. Since `FancyBorder` renders `{props.children}` inside a `<div>`, the passed elements appear in the final output.
 
 While this is less common, sometimes you might need multiple "holes" in a component. In such cases you may come up with your own convention instead of using `children`:
 
-```js{5,8,18,21}
-function SplitPane(props) {
-  return (
-    <div className="SplitPane">
-      <div className="SplitPane-left">
-        {props.left}
-      </div>
-      <div className="SplitPane-right">
-        {props.right}
-      </div>
-    </div>
-  );
-}
-
-function App() {
-  return (
-    <SplitPane
-      left={
-        <Contacts />
-      }
-      right={
-        <Chat />
-      } />
-  );
-}
-```
+`embed:composition-vs-inheritance/split-pane-example.js`
 
 [Try it on CodePen.](https://codepen.io/gaearon/pen/gwZOJp?editors=0010)
```

Comment:
Hello!

It's the first time I'm reading the code and I'm wondering.
Why don't we move this example to CodeSandbox?

Reply:
Yeah my intention was to move it to CodeSandbox. But in this codepen example, we have both css and javascript and I'm not sure about how to do that within a single javascript file in the `example` folder. Any thoughts?

Reply:
One question: Is it possible to reference multiple files through a  `codesandbox: //` link? If it is possible, I see that it is possible to solve the problem if we create a  `style.css` file and reference it in ` index.js`.

Reply:
Another idea would be to create `css-in-js`, but I'm afraid to cause some unnecessary 'complexity' in the documentation.

What you think about ?

Reply:
> Is it possible to reference multiple files through a codesandbox: // link?

I've tried that but it didn't work. Is it a gatsby plugin that we are using? Maybe should see how that works.

> Another idea would be to create css-in-js...

For the second approach, I think it could add much extraneous noise to the example snippets, which is meant to be clear and compact.